superset/sqllab/query_render.py

```diff
@@ -66,6 +66,19 @@ def render(self, execution_context: SqlJsonExecutionContext) -> str:
         except TemplateError as ex:
             self._raise_template_exception(ex, execution_context)
             return "NOT_REACHABLE_CODE"
+        except Exception as ex:
+            from superset.jinja_context import UndefinedTemplateFunctionException
+
+            if isinstance(ex, UndefinedTemplateFunctionException):
+                return query_model.sql.strip().strip(";")
+            raise
+
+    def _strip_sql_comments(self, sql: str) -> str:
+        import re
+
+        sql = re.sub(r"/\*.*?\*/", "", sql, flags=re.DOTALL)
+        sql = re.sub(r"--[^\n\r]*", "", sql)
+        return sql
```

Comment:
The regex approach is flaky, it can't handle nested comments (supported by some dialects) and might produce invalid SQL:

```python
>>> sql = "SELECT '--debug' AS flag"
>>> re.sub(r"--[^\n\r]*", "", sql)
"SELECT '"
```

Since you have a reference to the database it's better to use the SQL parser:

```python
dialect = execution_context.query.database.db_engine.spec.engine
sql_for_validation = SQLScript(sql, dialect).format(comments=False)
```